### ISSUE TITLE
Solver uses Name of the Elements without using Brand of the Elements

### ISSUE THREAD

**Issue by bloop428** [doxxx/ffxiv-craft-opt-web#295]
## Required Information

Class: Weaver
Level: 60
Craftsmanship: 661
Control: 619
CP: 377
Recipe Name: Holy Rainbow Cloth
Recipe Level: 52
Solver Seed: 87521

## Expected Behaviour

Name of the Elements is used in combination with Brand of the Elements and never on its own.

## Actual Behaviour

Name of the Elements is in the macro without Brand of the Elements to go along with it

## Steps To Reproduce

I am unsure if any of these settings are different from default so I will show all of them.
![image](https://user-images.githubusercontent.com/6138413/61996495-b65a8e80-b049-11e9-84bf-4f2b44375e4e.png)

**Comment by doxxx** [comment]
Name of the Elements has no effect on its own (in our model), so there's two possibilities:

1. It's a matter of timing -- i.e. the solver needs a "do nothing" action there. Do you have Use Conditions turned on in the Simulator options?
2. At some point it did have both but evolved the solution such that the Brand action got removed/replaced and the presence of the Name action did not penalize the new solution sufficiently to discard it.

That said, I don't think we're modeling the effect of Name of the Elements on Brand of the Elements correctly.

**Comment by bloop428** [comment]
Use Conditions was enabled during this

**Comment by doxxx** [comment]
The model has been improved recently, is this still an issue?

**Comment by j-ch** [comment]
Hope I can jump in here with more data. Seems like maybe the simulator has a general problem with preconditions?

```
Firefox 69.0.3
Seed: 87628

Crafter:
  Class: Weaver
  Level: 70
  Craftsmanship: 1621
  Control: 1574
  CP: 465
  Specialist: true

Recipe:
  Level: 290
  Difficulty: 2952
  Durability: 80
  Start Quality: 1661
  Max Quality: 13187
  Aspect: none

Settings:
  Max Tricks Uses: 0
  Reliability: 100%
  Use Conditions: true
  Algorithm: eaSimple
  Population: 300
  Generations: 100
  Penalty Weight: 10000

```

Which resulted in...

```
/ac "Inner Quiet" <wait.2>
/ac "Steady Hand II" <wait.2>
/ac "Prudent Touch" <wait.3>
/ac "Careful Synthesis III" <wait.3>
/ac "Steady Hand II" <wait.2>
/ac "Innovative Touch" <wait.3>
/ac "Patient Touch" <wait.3>
/ac "Patient Touch" <wait.3>
/ac "Advanced Touch" <wait.3>
/ac "Comfort Zone" <wait.2>
/ac "Careful Synthesis III" <wait.3>
/ac "Careful Synthesis II" <wait.3>
/ac "Great Strides" <wait.2>
/ac "Master's Mend II" <wait.3>
/echo Macro #1 complete <se.1>
/ac "Observe" <wait.3>
/ac "Focused Touch" <wait.3>
/ac "Careful Synthesis III" <wait.3>
/ac "Observe" <wait.3>
/ac "Focused Touch" <wait.3>
/ac "Careful Synthesis III" <wait.3>
/ac "Great Strides" <wait.2>
/ac "Specialty: Reflect" <wait.3>
/ac "Specialty: Reflect" <wait.3>
/ac "Byregot's Blessing" <wait.3>
/ac "Careful Synthesis III" <wait.3>
/echo Macro #2 complete <se.14>
```

**Comment by doxxx** [comment]
No, the model does account for Specialty: Reflect requiring Initial Preparation. In this case, if you look at the Monte Carlo Sim log you'll see that those two actions are "wasted" (the number in the WAC column increases).

Even with those two wasted actions, the sequence was still the best the solver could come up with during that run. If you continued to run the solver a few more times, it would probably find something better. There's a significant random aspect to the solver so these situations can occur.

I cannot stress this enough: if you don't like the sequence that the solver came up with, try "Continue" a few more times and if it's still no good, "Reset" and try again.
